### PR TITLE
Combo affidavit supplement

### PR DESCRIPTION
**Pull request by BryceStevenWilley**
Combines the Affidavit of Indigency with [its supplement](https://github.com/SuffolkLITLab/docassemble-AffidavitOfIndigency). I'll archive that repo when this PR is merged.

Changes to the affidavit interview, outside of just adding the supplement questions:

* added features to standalone affidavit that were in standalone supplement
    * sections
    * no digital signature requirement
* can double check if someone really needs supplement
  * if they give a household income that qualifies automatically, we can
    stop the rest of the supplement and go back to just one form
* asks all supplement questions before asking what fees they want waived
* split off review page
  * re-calcs if you added household members and other things that would change automatic eligibilty on the review page

### CODE REVIEW

File: docassemble/ALAffidavitOfIndigency/data/questions/affidavit_of_indigency_supplement_body.yml [hunk 1]
```diff
@@ -0,0 +1,1015 @@
...
+  users[0].nonemployment.moved
+  for income_type in users[0].nonemployment.elements:
+    income_type.complete
+  users[0].nonemployment.revisit = True
+  users[0].nonemployment.gathered = True
+---
+question: |
+  Tell us about your ${ ordinal(i) } income
+subquestion: |
+  For example, you might recieve disability benefits (SSDI) for $1000 once a month.
```
Comment (nonprofittechy) [inline docassemble/ALAffidavitOfIndigency/data/questions/affidavit_of_indigency_supplement_body.yml:320]:
```suggestion
```
This is confusing when it comes up in the context of another specific income type than SSDI.

**Review by nonprofittechy — CHANGES_REQUESTED** [review]
This is excellent--a very thoughtful interview. The only problem I see with the way it is currently set up is that it will ask the supplement questions for an existing integration, but a typical integration won't prompt downloading the supplement you just spent time completing.

I'd suggest making a new bundle to replace the existing single attachment so existing integrations will be able to download both forms. 

Happy to defer this until January or to take that task on though--just LMK implications for your projects of having a single PDF. Is that an issue for the Appeals Court e-filing, for example? Or is a single PDF combining both files OK?

**Comment by BryceStevenWilley** [comment]
Good catch, I should be able to change that. I'll see if it causes issues with already - running interviews (it might, changing the type of a variable in question files usually does), but I just miss that aspect.

Shouldn't cause any issue with the Affidavit: there's not a separate filing code, and my understanding is that tyler's "Attachment" feature hides information from judges by default, and some prefer to just have things in the same PDF. I can imagine that this would be a similar case, as it's only an additional 2 pages.

**Comment by BryceStevenWilley** [comment]
So replacing the variable will break existing interviews if the `affidavitofindigency_attachment` has already been created, anytime after the preview screen, and notably, it'll show an error on the final download screen for existing interviews. Sometimes I ran into a 500 error that didn't show our error-catcher screen too, which isn't great.

That seems like a deal-breaker for me, I don't want to break every interview that anyone ever used the affidavit in. I'll make a new variable for the combined affidavit, and make PRs for all of the other interviews this on is used in. Merging this won't be breaking, but will have a sub-optimal experience (as you pointed out, the supplement doc won't be downloaded), so we should release all of them at the same time.

**Comment by nonprofittechy** [comment]
> So replacing the variable will break existing interviews if the `affidavitofindigency_attachment` has already been created, anytime after the preview screen, and notably, it'll show an error on the final download screen for existing interviews. Sometimes I ran into a 500 error that didn't show our error-catcher screen too, which isn't great.
> 
> That seems like a deal-breaker for me, I don't want to break every interview that anyone ever used the affidavit in. I'll make a new variable for the combined affidavit, and make PRs for all of the other interviews this on is used in. Merging this won't be breaking, but will have a sub-optimal experience (as you pointed out, the supplement doc won't be downloaded), so we should release all of them at the same time.

An alternative is to add a `mandatory` code block (it's important that is has an `id`) that deletes any existing `affidavitofindigency_attachment` variable. I've used this many times when trying to refactor an object type without breaking existing sessions.

I also usually flag it for removal after a certain amount of time (maybe 6 months). But we could also wrap it in a `isinstance()` check.

**Comment by BryceStevenWilley** [comment]
That approach doesn't seem to be working for some reason.  If I pull the `main` branch of this repo into the playground, go to the ending screen (with or without supplement), then pull this branch (`combo_affidavit_supplement`) into same playground project and refresh the ending screen (simulating what would happen to existing interview sessions on an upgrade), I either get:
* a 504 timeout (refreshing after this gets the error catchall, but the error information isn't present, another downside of it)
* an infinite loop, between the ending screen and the `.enabled` attribute, which never seems to actually set? It's hard to tell

Unless you have an idea of how to fix it, I'm fairly stuck, and not sure how to progress with the same variable name.

**Comment by BryceStevenWilley** [comment]
I figured it out after getting more detailed stack traces from docassemble: `del` on the object itself doesn't delete the object inside the al_user_bundle or al_court_bundle. The infinite loop was happening because there were two objects (the original ALDocument and the new ALDocumentBundle) with the same instance name. Deleting the `al_user_bundle` and the `al_court_bundle` helped recreate everything.

I can confirm that the ALDocument object is only used in `al_user_bundle` and `al_court_bundle`; the MAVC version is used in other objects, but that's separate.

I need to do some more tests with other interviews and see if I can reproduce the 500 error at all, or if it was just related to this.

**Comment by BryceStevenWilley** [comment]
Tested within the Motion to Stay General, and works fine with existing sessions. Wasn't able to reproduce the 500, which did seem to be related to the infinite loop, though I don't understand why it wouldn't show our error catch screen instead, it's likely an upstream problem.

Should be good here, if you want to take one more look at it @nonprofittechy

**Review by nonprofittechy — APPROVED** [review]

**Pull request merged**